Modules/Core/Common/src/itkPoolMultiThreader.cxx

```diff
@@ -193,8 +200,16 @@ PoolMultiThreader ::ParallelizeArray(SizeValueType             firstIndex,
     // now wait for the other computations to finish
     for (SizeValueType i = 1; i < workUnit; i++)
     {
-      exceptionHandler.TryAndCatch([this, i, &reporter] {
-        m_ThreadInfoArray[i].Future.get();
+      exceptionHandler.TryAndCatch([this, i, &reporter, &filter] {
+        std::future_status status;
+        do
+        {
+          status = m_ThreadInfoArray[i].Future.wait_for(threadCompletionPollingInterval);
+          if (filter)
+          {
+            filter->IncrementProgress(0);
```

Comment:
This fine tweak to consider, but this method always trigger a ProgressReport event. There are some cases where regular updates may be nice ( as this should provide ) or situations where the performance of not emitting extra events would be desired.